### ISSUE TITLE
Rule Request: Prefer implicit member expression

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#3544]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Implicit member expressions make code nicer without much loss of readability.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
foo.color = NSColor.white

bar.color = NSColor.white.withAlphaComponent(0.5) // Swift 5.4
```

Would not trigger:

```swift
foo.color = .white

bar.color = .white.withAlphaComponent(0.5) // Swift 5.4
```

3. Should the rule be configurable, if so what parameters should be configurable?

No

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Up to you. I manually enable the rules I want.

**Comment by jpsim** [comment]
Rules like this one have been requested many times before, but there’s simply not enough information available just based on the syntax parsing alone to determine if it’s safe to omit a type prefix (`UIColor.white` -> `.white`).

This would need to be an analyzer rule, but those are significantly more difficult to configure on a project and take much longer to run. Would that be acceptable to you?

**Comment by sindresorhus** [comment]
> Would that be acceptable to you?

Yes
